lib/ansible/modules/service.py

```diff
@@ -1593,7 +1593,7 @@ def service_control(self):
             self.execute_command("%s %s %s" % (self.stopsrc_cmd, srccmd_parameter, self.name))
             srccmd = self.startsrc_cmd
 
-        if self.arguments and self.action == 'start':
+        if self.arguments and self.action == 'start' or self.arguments and self.action == 'restart':
```

Comment:
simpler:
``` if self.arguments and self.action in ('start', 'restart'):```

Also needs changelog and tests

Reply:
thought this might not work for most 'restarts', but for 'stop and start' versions of that

Reply:
Thanks for your feedback - very helpful!

The restart is parsed to stop and start one after another if I understand this correctly:

```python
        elif self.action == 'restart':
            self.execute_command("%s %s %s" % (self.stopsrc_cmd, srccmd_parameter, self.name))
            srccmd = self.startsrc_cmd
```